### PR TITLE
Updates the distributed CPU training documentation to add instructions for running on a Kubernetes cluster

### PR DESCRIPTION
**Pull request by dmsuehir**
# What does this PR do?

This PR updates the distributed CPU training documentation to include instructions for using multiple CPU nodes from a Kubernetes cluster. The instructions are kept pretty simple, assuming that the user already has access to a Kubernetes cluster and understands the basics of how to use it. The Kubernetes instructions build upon the existing distributed CPU example by using the same [question answering script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/question-answering) with [Kubeflow's PyTorch training operator](https://www.kubeflow.org/docs/components/training/pytorch/), while explaining along the way how to use Intel software optimizations and set the CPU/memory limits/requests in the specification file.

CC: @sywangyi @ashahba

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu and @MKhalusova

### CODE REVIEW

**Comment by sywangyi** [comment]
Thanks for adding practice in Kubernetes cluster.

**Review by sywangyi — APPROVED** [review]

File: docs/source/en/perf_train_cpu_many.md [hunk 1]
```diff
@@ -68,9 +69,9 @@ torch_ccl_path=$(python -c "import torch; import torch_ccl; import os;  print(os
 source $torch_ccl_path/env/setvars.sh
 ```
 
-#### IPEX installation:
+#### Intel® Extension for PyTorch installation:
```
Comment (stevhliu) [inline docs/source/en/perf_train_cpu_many.md:72]:
```suggestion
#### Intel® Extension for PyTorch installation
```

File: docs/source/en/perf_train_cpu_many.md [hunk 2]
```diff
@@ -68,9 +69,9 @@ torch_ccl_path=$(python -c "import torch; import torch_ccl; import os;  print(os
...
 
-IPEX provides performance optimizations for CPU training with both Float32 and BFloat16, you could refer [single CPU section](./perf_train_cpu).
+Intel Extension for PyTorch (IPEX) provides performance optimizations for CPU training with both Float32 and BFloat16, you could refer [single CPU section](./perf_train_cpu).
```
Comment (stevhliu) [inline docs/source/en/perf_train_cpu_many.md:74]:
```suggestion
Intel Extension for PyTorch (IPEX) provides performance optimizations for CPU training with both Float32 and BFloat16 (refer to the [single CPU section](./perf_train_cpu) to learn more).
```

**Review by stevhliu — APPROVED** [review]
Thanks for adding instructions for Kubernetes! cc @ArthurZucker for a quick final review since I'm not too familiar with this topic :)

**Comment by dmsuehir** [comment]
@stevhliu Thank you for the review, I made the updates that you suggested.

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗

**Pull request merged**